### DISCUSSION TITLE
How to transfer from localhost to server?

### DISCUSSION THREAD

**Discussion by pstephenson5** [typemill/typemill#439] [Q&A]
I'm built a site on my localhost but when I transfer it to a web host using SFTP all the URLs redirect to the localhost address. I  did a search and replace on the entire codebase replacing the localhost URL and port with the hosted domain but the issue still persists. What is the process for transferring a site from localhost to a web host? I couldn't find this documented anywhere.

**Comment by trendschau** [comment]
The Navigation is cached and contains the old url. To recreate the cache, simply  unpublish and publish a page again or delete the cached  files in data/navigation. 

Next version will include the delete cache feature again.

**Reply by trendschau** [reply]
I will check the documentation again

**Reply by pstephenson5** [reply]
I tried deleting the cached files in data/naviagtion on the web host. This made no difference. Somewhere, somehow Typemill is using my internal localhost IP address for all URLs other than the homepage.

**Reply by trendschau** [reply]
Can you provide some more details?

* Is the navigation the problem or the links in the content-area (images, files, internal links)?
* Can you access the correct url directly?
* Do you have any redirect rules in the htaccess-file?
* Do you use a proxy setup, Docker or something like that? Or subdomains? Or a simple domain like https://mywebsite.io
* Do you have mod_rewrite module active? 

The only place where Typemill stores absolute urls is the cached navigation, so it is very hard from the distance to check that problem. You can also send me a link via mail so I can check it and try to help: https://typemill.net/info/imprint-and-privacy

**Reply by pstephenson5** [reply]
Thank you - I have emailed you with more details.

**Comment by trendschau** [comment]
Updated the documentation: https://typemill.net/getting-started/installation/localhost
